File: C#/BluffinMuffin.Protocol/Lobby/RegisteredMode/AuthenticateUserCommand.cs
using BluffinMuffin.Protocol.Enums;
using Com.Ericmas001.Net.Protocol;

namespace BluffinMuffin.Protocol.Lobby.RegisteredMode
{
    /// <summary>
    /// This command will try to authenticate the client against the user database on the server.
    /// </summary>
    public class AuthenticateUserCommand : AbstractLobbyCommand, ICommandWithResponse<AuthenticateUserResponse>
    {
        /// <summary>
        /// The username
        /// </summary>
        [ExampleValue("ericmas001")]
        public string Username { get; set; }
        /// <summary>
        /// The password associated with the username
        /// </summary>
        [ExampleValue("0nc3Up0nAT1m3")]
        public string Password { get; set; }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public AuthenticateUserResponse Response(bool success, BluffinMessageId msgId, string message)
        {
            return new AuthenticateUserResponse(this) { Success = success, MessageId = msgId, Message = message };
        }
    }
}

File: Python/BluffinMuffin.Protocol/lobby/registered_mode/authenticate_user_command.py
from abstract_lobby_command import AbstractLobbyCommand

__author__ = 'ericmas001@gmail.com'



class AuthenticateUserCommand(AbstractLobbyCommand):
    def __init__(self, obj):
        super().__init__(obj)
        self.username = obj['Username']
        self.password = obj['Password']

    def __str__( self ):
        return '{0} ({1} : {2})'.format(super().__str__(), self.username, self.password)
